packages/ds-healthcare-gov/src/components/Header/ActionMenu.tsx

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable react/no-multi-comp, jsx-a11y/no-redundant-roles */
+import { Button, CloseIconThin, MenuIcon } from '@cmsgov/design-system';
+import PropTypes from 'prop-types';
+import React, { SyntheticEvent } from 'react';
+import classnames from 'classnames';
+import { sendHeaderEvent } from './analytics';
+import { TFunction } from 'react-i18next';
+
+const menuId = 'hc-c-menu';
+
+export interface LinkProps {
+  href: string;
+  label: React.ReactNode;
+  onClick?: (event: SyntheticEvent) => any;
+}
+
+export interface ActionMenu {
+  t: TFunction;
+  /** Applies the inverse theme styling */
+  firstName?: string;
+  loggedIn?: boolean;
+  onMenuToggleClick: (event: SyntheticEvent) => any;
+  /**
+   * Indicates the menu is open, which influences the label
+   * and ARIA attribute of the toggle button.
+   */
+  open?: boolean;
+  /**
+   * These are the links that are visible in the upper left when there
+   * is no menu button present. Currently, these only show up when the
+   * user is logged out
+   */
+  links: LinkProps[];
+}
+
+/**
+ * ActionMenu is displayed at the top-right of the header.
+ * There are two variations of this component: One for logged-out
+ * users, which displays a "Log in" & "Espanol" link on desktop.
+ * And another for logged-in users, which includes the user's first
+ * name. The logged-in variation always displays a "Menu" toggle
+ * button, and the logged-out variation only displays it on mobile.
+ */
+const ActionMenu = function (props) {
```

Comment:
Did these props not need to have the type `ActionMenu`?

Reply:
Yup

Reply:
Hooking this up properly spawned a few more TS errors that I'm now working through 👍